Automattic/jetpack
# jetpack: sharing: Remove functions deprecated in Jetpack 11.0

## Proposed changes:

These functions were deprecated in #24040, which was included in Jetpack 11.0 released June 2022.

The `Jetpack_ReCaptcha` class is still in use on wpcom, so leaving that in place for now.

### Other information:


## Jetpack product discussion


p1723651246448749-slack-C029E4HPT

## Does this pull request change what data or activity we track or use?


When this code was in use it needed to be mentioned on the privacy page, but it hasn't been in use for two years. p1723651246448749-slack-C029E4HPT is a request to remove this old code entirely before removing the relevant entries from that page.

## Testing instructions:






* Make sure stuff still works.
* Check that wpcom doesn't use any of these (I already did, but double-checking is good).
